src/view/RoomsView.java

```diff
@@ -164,6 +167,19 @@ public void keyReleased(KeyEvent e) {}
           }
         });
 
+    settingsButton.addActionListener(
+        evt -> {
+          if (evt.getSource().equals(settingsButton)) {
+            RoomsState currentState = viewModel.getState();
+            String roomUid = currentState.getRoomUid();
+            for (var room : currentState.getAvailableRooms()) {
```

Comment:
We are trying to get rid of this pattern. You should make use of the null checker and get Room by uid logic in `RoomsState`.